### DISCUSSION TITLE
how to get commission of each trade.

### DISCUSSION THREAD

**Discussion by sultank1** [kernc/backtesting.py#707] [Q&A] [answered]
hey @kernc , is there any way to add a new column(commission) in 
`stat._trades
`

want to verify if the commissions are being calculated correctly...

by the way till now i thought that commission parameter was in percentage and while reading documentation i found out i was wrong...  

Learned the lesson the HARD way, to always read the documentations first ...

moreover i cant even explain how much have you helped me with your awesome library.
I pray for your good health and happiness.

**Comment by kernc** [comment, marked as answer]
> any way to add a new column(commission) in `stat._trades `
> want to verify if the commissions are being calculated correctly...

You can compare the trade EntryPrice with its respective bar's Open price.

See what you can do with the following piece of code:
```py
trades = stats['_trades'].set_index('EntryTime')
prices = data.Open[trades.index]
commissions = (prices - trades.EntryPrice) * -np.sign(trades.Size) / prices
```
E.g. `commissions` with [Quick Start example](https://kernc.github.io/backtesting.py/doc/examples/Quick%20Start%20User%20Guide.html#Trade-data):
```text
2004-09-17    0.002
2004-11-12    0.002
2004-11-26    0.002
2004-11-30    0.002
2004-12-06    0.002
              ...  
2012-11-29    0.002
2013-01-08    0.002
2013-01-10    0.002
2013-01-23    0.002
2013-01-31    0.002
Length: 153, dtype: float64
```

**Reply by sultank1** [reply]
> > any way to add a new column(commission) in `stat._trades `
> > want to verify if the commissions are being calculated correctly...
> 
> You can compare the trade EntryPrice with its respective bar's Open price.
> 
> See what you can do with the following piece of code:
> 
> ```python
> trades = stats['_trades'].set_index('EntryTime')
> prices = data.Open[trades.index]
> commissions = (prices - trades.EntryPrice) * -np.sign(trades.Size) / prices
> ```
> 
> E.g. `commissions` with [Quick Start example](https://kernc.github.io/backtesting.py/doc/examples/Quick%20Start%20User%20Guide.html#Trade-data):
> 
> ```
> 2004-09-17    0.002
> 2004-11-12    0.002
> 2004-11-26    0.002
> 2004-11-30    0.002
> 2004-12-06    0.002
>               ...  
> 2012-11-29    0.002
> 2013-01-08    0.002
> 2013-01-10    0.002
> 2013-01-23    0.002
> 2013-01-31    0.002
> Length: 153, dtype: float64
> ```

thanks

**Comment by Chris2L** [comment]
Would the correct way to calculate the total spent on commissions be the following:

`    trades = stat['_trades'].set_index('EntryTime')
    prices = data.Open[trades.index]
    commissions = (prices - trades.EntryPrice) * -np.sign(trades.Size) * abs(trades.Size)
    print(commissions.sum())`

Basically the difference in price times the size of the order?
